### PR TITLE
FM2-484: Fix Diagnostic Report thymeleaf template expression 

### PR DESCRIPTION
**Pull request by pajmera4**
## Description of what I changed

This PR contains changes to fix template parsing error while creating narratives for DiagnosticReport. The error comes when attachment.url is present in the resource. 

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-484

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`
Existing test cases are updated to include the use case for passing url in presentedForm

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by pajmera4** [comment]
The Pr is raised against this issue: https://issues.openmrs.org/browse/FM2-484

File: test-data/src/test/resources/org/openmrs/module/fhir2/narrative/example/diagnosticreport-example.json [hunk 1]
```diff
@@ -66,5 +66,9 @@
...
     }
   ],
-  "conclusion": "Core lab"
-}
+  "conclusion": "Core lab",
+  "presentedForm": [{
+    "url": "labReport.pdf",
+    "title": "Lab Report"
+  }]
+}
```
Comment (ibacher) [inline test-data/src/test/resources/org/openmrs/module/fhir2/narrative/example/diagnosticreport-example.json:74]:
```suggestion
}

```

File: test-data/src/test/resources/org/openmrs/module/fhir2/narrative/expected/diagnosticreport-expected.html [hunk 2]
```diff
@@ -1 +1 @@
...
...quest/req">ServiceRequest/req</a></td></tr><tr><td>Status:</td><td>FINAL</td></tr><tr><td>Category:</td><td> Haematology test </td></tr><tr><td>Code:</td><td> Complete blood count (hemogram) panel - Blood by Automated count </td></tr><tr><td>Subject:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/Patient/f001">P. van den Heuvel</a></td></tr><tr><td>Issued:</td><td>Wed May 15 18:32:52 UTC 2013</td></tr><tr><td>Performer:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/Organization/f001">Burgers University Medical Centre</a></td></tr><tr><td>Result:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f001">Observation/f001</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f002">Observation/f002</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f003">Observation/f003</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f004">Observation/f004</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f005">Observation/f005</a></td></tr><tr><td>Conclusion:</td><td>Core lab</td></tr><tr><td>Presented Form:</td><td><a href="labReport.pdf">Lab Report</a></td></tr></tbody></table></div>
```
Comment (ibacher) [inline test-data/src/test/resources/org/openmrs/module/fhir2/narrative/expected/diagnosticreport-expected.html:1]:
```suggestion
<div xmlns="http://www.w3.org/1999/xhtml"><table class="hapiPropertyTable"><tbody><tr><td>Id:</td><td>DiagnosticReport/f001</td></tr><tr><td>Identifier:</td><td><div>nr1239044</div></td></tr><tr><td>Based On:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/ServiceRequest/req">ServiceRequest/req</a></td></tr><tr><td>Status:</td><td>FINAL</td></tr><tr><td>Category:</td><td> Haematology test </td></tr><tr><td>Code:</td><td> Complete blood count (hemogram) panel - Blood by Automated count </td></tr><tr><td>Subject:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/Patient/f001">P. van den Heuvel</a></td></tr><tr><td>Issued:</td><td>Wed May 15 18:32:52 UTC 2013</td></tr><tr><td>Performer:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/Organization/f001">Burgers University Medical Centre</a></td></tr><tr><td>Result:</td><td><a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f001">Observation/f001</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f002">Observation/f002</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f003">Observation/f003</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f004">Observation/f004</a> , <a href="http://localhost:8080/openmrs/ws/fhir2/R4/Observation/f005">Observation/f005</a></td></tr><tr><td>Conclusion:</td><td>Core lab</td></tr><tr><td>Presented Form:</td><td><a href="labReport.pdf">Lab Report</a></td></tr></tbody></table></div>

```

**Comment by ibacher** [comment]
@pajmera4 Thanks for this. A few process notes:

1. We prefer that commits have the ticket number included in the commit so, when reviewing the history we can trace back why the change was made.
2. When creating a PR, it's good to add a link from the Jira card to the PR.

**Comment by pajmera4** [comment]
> 1. umber included in the commit so,



> 

Thanks @ibacher. Sure, will take care!!

**Pull request merged**